src/mbgl/map/tile_data.cpp

```diff
@@ -57,7 +58,19 @@ void TileData::cancel() {
     }
 }
 
+bool TileData::mayStartParsing() {
+    return !parsing.test_and_set(std::memory_order_acquire);
```

Comment:
> Why is this necessary?

Because tiles in the `partial` state are always parsed, this will prevent the same tile from being parsed in parallel by one or more workers.